pkg/controller/clone/planner.go

```diff
@@ -367,14 +377,14 @@ func (p *Planner) computeStrategyForSourceSnapshot(ctx context.Context, args *Ch
 		return nil, nil
 	}
 
-	// Do snapshot and storage class validation
-	targetStorageClass, err := GetStorageClassForClaim(ctx, p.Client, args.TargetClaim)
-	if err != nil {
-		return nil, err
-	}
-	if targetStorageClass == nil {
-		return nil, fmt.Errorf("target claim's storageclass doesn't exist, clone will not work")
+	if targetStorageClass.VolumeBindingMode != nil &&
+		*targetStorageClass.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer &&
+		cc.ImmediateBindingRequested(args.TargetClaim) {
+		args.Log.V(3).Info("Immediate binding requested, need to fall back to host assisted")
```

Comment:
This should not be necessary.  The prep phase can bind pvc' to a random node.

Reply:
you mean leave without the fallback and create the pvc' without waiting for selected node and it will just get bound  with the pod of the clone? or need to implement something for that?

Reply:
Yeah shouldn't have to make any changes to the planner

Reply:
done